views/detail_views.py

```diff
@@ -1,52 +1,87 @@
-import pandas as pd
 
 
 class MasterDetailView(DetailView):
     mode = None
     template_name = 'stemp_abw/popups/base_layer_popup.html'
     context_object_name = 'layer'
 
-    def get_context_data(self, **kwargs):
-        context = super(MasterDetailView, self).get_context_data(**kwargs)
-
-        context['title'] = LABELS['layers'][self.model.name]['title']
-        context['text'] = LABELS['layers'][self.model.name]['text']
-
-        for layer_group in LAYER_REGION_METADATA.values():
+    def get_source_data(self, metadata, app_name):
+        """
+        This method takes a metadata ConfigObj and returns a list
+        with 0 OR n-amount of Source objects, if primary keys (PK)s
+        of sources records in database are provided in ConfigObj object.
+        Values in the metadata config file should correspond to PKs
+        as list of values (1,2,3,...n). if the sole value 0 is provided
+        in the metadata config file then the returned list is empty.
+
+        Parameters
+        ----------
+        metadata : :obj:`ConfigObj`
+        app_name : :obj:`str`
+
+        Returns
+        -------
+        :obj:`list` of :obj:`Source`
+            List with 0 OR n-amount of Source objects.
+        """
+        for layer_group in metadata.values():
             for layer in layer_group.values():
                 if layer['model'] == self.model.name:
                     sources = []
                     for source in layer['sources']:
                         if source == '0':
-                            sources.append(source)
                             break
                         else:
-                            sources.append(Source.objects.filter(app_name='stemp_abw').get(pk=source))
-                    context['sources'] = sources
-        for layer_group in LAYER_AREAS_METADATA.values():
-            for layer in layer_group.values():
-                if layer['model'] == self.model.name:
-                    sources = []
-                    for source in layer['sources']:
-                        if source == '0':
-                            sources.append(source)
-                            break
-                        else:
-                            sources.append(Source.objects.filter(app_name='stemp_abw').get(pk=source))
-                    context['sources'] = sources
+                            sources.append(Source.objects
+                                           .filter(app_name=app_name)
+                                           .get(pk=source))
+                    return sources
+                else:
+                    pass
+
+    def get_context_data(self, **kwargs):
+        context = super(MasterDetailView, self).get_context_data(**kwargs)
+
+        context['title'] = LABELS['layers'][self.model.name]['title']
+        context['text'] = LABELS['layers'][self.model.name]['text']
+
+        # Get app_name from request
+        app_name = self.request.resolver_match.app_name
```

Comment:
Why not using the `APP_NAME` from app_settings.py here? (see above)

Reply:
See answer above.